app/models/concerns/hyrax/ability.rb

```diff
@@ -46,7 +54,7 @@ def download_users(id)
     def can_create_any_work?
       Hyrax.config.curation_concerns.any? do |curation_concern_type|
         can?(:create, curation_concern_type)
-      end && source_ids_for_deposit.any?
+      end && Hyrax::Collections::PermissionsService.admin_set_ids_for_user(user: current_user, access: ['deposit', 'manage'], ability: self).any?
```

Comment:
Before CE work, the original check here was `&& admin_set_ids_for_deposit.any?`.  This method in the abilities class queried the permission template access table to see if the current_user had deposit access to any admin set.  If they can't deposit in an admin set, then they can't add works even if they pass `can?(:create, curation_concern_type)` for some work type.

With the CE change to store admin set and collection permissions in the same table, there needed to be a specific method that gets the list of ids for admin sets only in which the user can deposit since admin sets require membership.  

The actual 'chain' processing here is the same.  
* the first part is checking whether the user can create a work of one of the work types (`.any?` means this will return `true | false`)
* the second part is checking whether the user can deposit in any admin sets by getting the list of ids (`.any?` means this will return `true | false`)
* the 'chaining' is simply a boolean && where both have to return true for `can_create_any_work?` to be true.

I'm not sure what else you are looking for in terms of 'Can you create a specific method on the PermissionsService class that better explains what is happening'.  I can add comments to this method here to explain this logic if you like.

Reply:
Thank you for the response. Your explanation helps clarify. I was reading the code through "Law of Demeter" lense (e.g. avoid chaining methods). However, a caveat is that you can use simple query methods at the end of a method (e.g. `.present?`, `.any?`). My personal preference is still to encode those questions as methods on the receiving object, thus tightening up the interface of the object.

However, I stand by your reasons. Consider this request "resolved"

Reply:
👍